action/action.go

```diff
@@ -33,9 +33,7 @@ type (
 	}
 
 	actionPayload interface {
-		Cost() (*big.Int, error)
 		IntrinsicGas() (uint64, error)
-		SetEnvelopeContext(*AbstractAction)
```

Comment:
now it will use `AbstractAction` in the `envelope`, not in each action, so this is no longer needed

Reply:
`Cost` may serve as an optional implementation method. If the payload is not implemented, it will be `price * intrinsicGas`; if implemented, the result implemented by the payload will be used.

For example, `execution` should be `price * gasLimit`, and `CreateStake` should be `price * intrinsicGas + stake amount`.

Reply:
`Cost` is implemented by every action, and used by actpool
now the actions (like `Execution`, `CreateStake`) does not have the `gasPrice` field, so have to move the implementation up to the `envelope` struct

Reply:
I see, but I prefer define an interface for payload to calc cost in envelop side, instead of calc by type explicitly